homeassistant/components/lutron_caseta/config_flow.py

```diff
@@ -31,22 +49,129 @@ class LutronCasetaFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
     def __init__(self):
         """Initialize a Lutron Caseta flow."""
         self.data = {}
+        self.lutron_id = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            self.data[CONF_HOST] = user_input[CONF_HOST]
+            return await self.async_step_link()
+
+        return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA_USER)
+
+    async def async_step_zeroconf(self, discovery_info):
+        """Handle a flow initialized by zeroconf discovery."""
+        hostname = discovery_info[ATTR_HOSTNAME]
+        if hostname is None or not hostname.startswith("lutron-"):
+            return self.async_abort(reason="not_lutron_device")
+
+        lutron_id = hostname.split("-")[1]
+        if lutron_id.endswith(".local."):
+            lutron_id = lutron_id[:-7]
+        self.lutron_id = lutron_id
```

Comment:
This could be compressed using a string replace
```suggestion
        self.lutron_id = lutron_id.replace(".local.", "")
```

Reply:
It shouldn't be possible for `.local.` to appear anywhere in the hostname except for the end (otherwise rfc violation and probably everything breaks) so this should be safe to do 👍